idseq_dag/steps/run_star.py

```diff
@@ -1,12 +1,218 @@
+import multiprocessing
+import os
+
 from idseq_dag.engine.pipeline_step import PipelineStep
 import idseq_dag.util.command as command
+import idseq_dag.util.s3 as s3
+import idseq_dag.util.log as log
 
 
 class PipelineStepRunStar(PipelineStep):
+    max_input_reads = 75 * 1000 * 1000
+
     def run(self):
-        input_file_0 = self.input_files_local[0][0]
-        input_file_1 = self.input_files_local[0][1]
-        output_file_0 = self.output_files_local()[0]
-        output_file_1 = self.output_files_local()[1]
-        command.execute("gzip -dc %s |head -20 > %s" % (input_file_0, output_file_0))
-        command.execute("gzip -dc %s |head -20 > %s" % (input_file_1, output_file_1))
+        """Run STAR to filter out host reads."""
+        # Setup
+        input_files = self.input_files_local[0][0:2]
+        star_genome = self.additional_files["star_genome"]
+        ref_dir = self.ref_dir_local
+        scratch_dir = os.path.join(self.output_dir_local, "scratch")
+        total_counts_from_star = {}
+        output_files_local = self.output_files_local()
+        num_inputs = len(self.input_files[0])
+
+        # Check genome file. Ex: s3://...STAR_genome.tar -> /ref/STAR_genome
+        basename = os.path.basename(star_genome).rstrip(".tar")
+        genome_dir = os.path.join(ref_dir, basename)
+        assert os.path.exists(genome_dir)
```

Comment:
Do we need these asserts? 
Best practices for asserts is for cases that should never happen. On the other hand, we should probably use a `try:  ... except:` clause instead. It will also allow use of custom exception classes.

Also, do not forget that asserts are not considered when optimization is requested at compilation time (https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement)